src/System.Runtime.Numerics/tests/ComplexTests.cs

```diff
@@ -301,23 +301,28 @@ public static void ACos_Basic(double real, double imaginary)
 
         public static IEnumerable<object[]> ACos_Advanced_TestData()
         {
+            // Simple values
+            yield return new object[] { 0.0, 0.0, Math.PI / 2.0, 0.0 };
+            yield return new object[] { 1.0, 0.0, 0.0, 0.0 };
+            yield return new object[] { -1.0, 0.0, Math.PI, 0.0 };
+            yield return new object[] { 0.0, 1.0, Math.PI / 2.0, -Math.Log(1.0 + Math.Sqrt(2.0)) };
+            yield return new object[] { 0.0, -1.0, Math.PI / 2.0, Math.Log(1.0 + Math.Sqrt(2.0)) };
+
             yield return new object[] { 1234000000, 0, 0, 21.62667394298955 }; // Real part is positive, imaginary part is 0
             yield return new object[] { 0, -1234000000, 1.5707963267948966, 21.62667394298955 }; // Imaginary part is negative
 
-            // Boundary values
-            yield return new object[] { double.MaxValue, 0, double.NaN, double.NaN };
-            yield return new object[] { double.MinValue, 0, double.NaN, double.NaN };
+            // Extreme values
+            yield return new object[] { double.MaxValue, 0.0, 0.0, Math.Log(2.0) + Math.Log(double.MaxValue) };
+            yield return new object[] { 0.0, double.MaxValue, Math.PI / 2.0, -(Math.Log(2.0) + Math.Log(double.MaxValue)) };
+            yield return new object[] { -double.MaxValue, -double.MaxValue, 3.0 / 4.0 * Math.PI, Math.Log(2.0 * Math.Sqrt(2.0)) + Math.Log(double.MaxValue) };
+
+            // NaN values
+            yield return new object[] { double.NaN, double.NaN, double.NaN, double.NaN };
+            yield return new object[] { -1.0, double.NaN, double.NaN, double.NaN };
+            yield return new object[] { double.NegativeInfinity, double.NaN, double.NaN, double.NaN };
+            yield return new object[] { double.NaN, 0.0, double.NaN, double.NaN };
+            yield return new object[] { double.NaN, double.PositiveInfinity, double.NaN, double.NaN };
 
-            // Invalid values
```

Comment:
We should preserve this old behavior in a "Legacy" test case, like we did for the `Tan` and `Tanh` test data. It should only run on .NET Framework.